.github/workflows/build.yaml

```diff
@@ -103,12 +103,13 @@ jobs:
           if-no-files-found: warn
           retention-days: 7
       - name: 'Run UI integration tests'
+        id: run_tests
         working-directory: ./liberty-tools-intellij
         run: bash ./src/test/resources/ci/scripts/run.sh
       - name: 'Archive Test logs and reports'
-        if: ${{ failure() }}
-        uses: actions/upload-artifact@v3
+        if: ${{ failure() && steps.run_tests.conclusion == 'failure' }}
```

Comment:
Please add details of 'steps.run_tests.conclusion' condition in a comment.

Reply:
I have added an extra condition to the step `Archive Test logs and reports` because, the test report is generated only in the previous step, `Run UI integration tests`. We need to execute the `Archive Test logs and reports` step only if the previous step (`Run UI integration tests`) fails. So that we can avoid this below warning,

![355503232-a625030d-0b88-49b5-bbaa-37fe0accbc99](https://github.com/user-attachments/assets/97126bac-2d30-4ed0-82c8-6b97207863d7)